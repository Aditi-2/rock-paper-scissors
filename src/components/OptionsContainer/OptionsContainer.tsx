import styled from "@emotion/styled";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import React from "react";
import { GameChoice, PlayerChoice } from "../../common/types/game";
import { HANDS } from "../../utils/common";

interface OptionsContainerProps {
  choices: GameChoice;
  handleChoiceSelection: (args: PlayerChoice) => void | Promise<void>
}

export const OptionsContainer: React.FC<OptionsContainerProps> = ({choices, handleChoiceSelection}) => {
  return (
    <ChoiceContainerStyled>
      {HANDS.map((hand, index) => (
        <ChoiceIconContainerStyled
          className="choices player-choice"
          key={hand.name}
        >
          <FontAwesomeIcon
            icon={hand.icon}
            className={`choice-icon ${
              choices.playerOne === null ? "player-choice-icon" : "disabled"
            }`}
            onClick={
              choices.playerOne === null
                ? () => handleChoiceSelection(index as PlayerChoice)
                : undefined
            }
          />
        </ChoiceIconContainerStyled>
      ))}
    </ChoiceContainerStyled>
  );
};

const ChoiceContainerStyled = styled.div`
  display: flex;
  width: 40%;
  margin: 0 auto;
  padding: 2rem 0;
`;

const ChoiceIconContainerStyled = styled.div`
  font-size: 35px;
  margin: 0 auto;
  color: #d6d6d6;
  @media only screen and (min-width: 800px) {
    font-size: 70px;
  }
  &.player-choice {
    width: 100%;
    @media only screen and (min-width: 800px) {
      width: 33%;
    }
    & .disabled {
      color: #ababab;
    }
    & .player-choice-icon {
      &:hover {
        color: #3f51b5;
      }
    }
  }
`;
