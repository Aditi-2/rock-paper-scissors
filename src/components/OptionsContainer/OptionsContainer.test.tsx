import React from 'react';
import {create} from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { OptionsContainer } from './OptionsContainer';

const handleChoiceSelection = jest.fn();

const getComponent = () => <OptionsContainer choices={{playerOne: null, playerTwo: null }} handleChoiceSelection={handleChoiceSelection}/>

test('OptionsContainer Component snapshot', () => {
    const component = create(getComponent()).toJSON();
    expect(component).toMatchSnapshot();
});

test('renders OptionsContainer Component', () => {
  render(getComponent());
  const images = screen.getAllByRole('img', {hidden: true});
  images.forEach(image => expect(image).toBeInTheDocument())
});

test('OptionsContainer buttons register click', () => {
    render(getComponent());
    const images = screen.getAllByRole('img', {hidden: true});
    images.forEach(image => fireEvent.click(image))
    expect(handleChoiceSelection).toHaveBeenCalledTimes(images.length)
  });
