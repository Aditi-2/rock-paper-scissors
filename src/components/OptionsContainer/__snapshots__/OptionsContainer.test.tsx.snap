// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsContainer Component snapshot 1`] = `
<div
  className="css-1r1j74i"
>
  <div
    className="choices player-choice css-1hy65oz"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-hand-rock fa-w-16 choice-icon player-choice-icon"
      data-icon="hand-rock"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464.8 80c-26.9-.4-48.8 21.2-48.8 48h-8V96.8c0-26.3-20.9-48.3-47.2-48.8-26.9-.4-48.8 21.2-48.8 48v32h-8V80.8c0-26.3-20.9-48.3-47.2-48.8-26.9-.4-48.8 21.2-48.8 48v48h-8V96.8c0-26.3-20.9-48.3-47.2-48.8-26.9-.4-48.8 21.2-48.8 48v136l-8-7.1v-48.1c0-26.3-20.9-48.3-47.2-48.8C21.9 127.6 0 149.2 0 176v66.4c0 27.4 11.7 53.5 32.2 71.8l111.7 99.3c10.2 9.1 16.1 22.2 16.1 35.9v6.7c0 13.3 10.7 24 24 24h240c13.3 0 24-10.7 24-24v-2.9c0-12.8 2.6-25.5 7.5-37.3l49-116.3c5-11.8 7.5-24.5 7.5-37.3V128.8c0-26.3-20.9-48.4-47.2-48.8z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="choices player-choice css-1hy65oz"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-hand-paper fa-w-14 choice-icon player-choice-icon"
      data-icon="hand-paper"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M408.781 128.007C386.356 127.578 368 146.36 368 168.79V256h-8V79.79c0-22.43-18.356-41.212-40.781-40.783C297.488 39.423 280 57.169 280 79v177h-8V40.79C272 18.36 253.644-.422 231.219.007 209.488.423 192 18.169 192 40v216h-8V80.79c0-22.43-18.356-41.212-40.781-40.783C121.488 40.423 104 58.169 104 80v235.992l-31.648-43.519c-12.993-17.866-38.009-21.817-55.877-8.823-17.865 12.994-21.815 38.01-8.822 55.877l125.601 172.705A48 48 0 0 0 172.073 512h197.59c22.274 0 41.622-15.324 46.724-37.006l26.508-112.66a192.011 192.011 0 0 0 5.104-43.975V168c.001-21.831-17.487-39.577-39.218-39.993z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="choices player-choice css-1hy65oz"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-hand-scissors fa-w-16 choice-icon player-choice-icon"
      data-icon="hand-scissors"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M216 440c0-22.092 17.909-40 40-40v-8h-32c-22.091 0-40-17.908-40-40s17.909-40 40-40h32v-8H48c-26.51 0-48-21.49-48-48s21.49-48 48-48h208v-13.572l-177.551-69.74c-24.674-9.694-36.818-37.555-27.125-62.228 9.693-24.674 37.554-36.817 62.228-27.124l190.342 74.765 24.872-31.09c12.306-15.381 33.978-19.515 51.081-9.741l112 64A40.002 40.002 0 0 1 512 168v240c0 18.562-12.77 34.686-30.838 38.937l-136 32A39.982 39.982 0 0 1 336 480h-80c-22.091 0-40-17.908-40-40z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
